torch/_dynamo/variables/ctx_manager.py

```diff
@@ -839,6 +837,39 @@ def fn_name(self):
         return "nullcontext"
 
 
+class ProfilerContextVariable(ContextWrappingVariable):
+    """
+    This class represents a set of torch profiler context objects, where Dynamo
+    ignores all the side-effects in the __init__, __enter__ and __exit__ methods
+    by treating the object mostly as a `contextlib.nullcontext`, except for edge
+    cases like the `__enter__` method which returns the object itself rather
+    than `None`, per implementation of the torch objects.
+    """
+
+    def __init__(self, **kwargs) -> None:
+        super().__init__(target_values=None, **kwargs)
+
+    def enter(self, tx):
+        return self
+
+    def exit(self, tx: "InstructionTranslator", *args):
+        return variables.ConstantVariable.create(None)
+
+    def module_name(self):
+        return "contextlib"
+
+    def fn_name(self):
+        return "nullcontext"
```

Comment:
Don't think these are right?

Reply:
I don't know what these are used for, but it should be possible to thread the right values through here when constructing one of these

Reply:
Again intentional, and it looks super sketchy indeed, but in a sense it's preserving the historical behavior, see [this comment](https://github.com/pytorch/pytorch/pull/145537/files#diff-39a2b39a335f1cac6506a29fb8618dd703c8130784dd6a6c2fd56adec8acd5a2R843-R844).

To be frank, we might need to fix something in this historical behavior as well, but that feels like a _big_ rabbit hole, so I'm just taking a small step here.

Reply:
fair enough